src/transformers/processing_utils.py

```diff
@@ -226,6 +226,11 @@ def _get_arguments_from_pretrained(cls, pretrained_model_name_or_path, **kwargs)
             args.append(attribute_class.from_pretrained(pretrained_model_name_or_path, **kwargs))
         return args
 
+    @property
+    def model_input_names(self):
+        first_attribute = getattr(self, self.attributes[0])
+        return getattr(first_attribute, "model_input_names", None)
```

Comment:
This is what the 'generic' property method would look like - we'd only then have to override this method for CLIP-like models that take inputs from the feature extractor and tokenizer. We wouldn't have to add the method for any speech models or single input domain vision models.

Reply:
Sounds good to me!